addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -315,54 +318,81 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
             resolveFadeOut();
         });
 
-        var nextQuestionPromise = this._rpc({
-            route: _.str.sprintf('/survey/session/next_question/%s', self.surveyAccessToken),
-            params: {
-                'go_back': goBack,
-            }
-        });
+        var refreshBackground = this.reloadBackgroundOnNext || goBack;
+        if (refreshBackground) {
+            $('div#wrapwrap').css("box-shadow", "inset 0 0 0 10000px rgba(255,255,255,1)");
+        }
 
         // avoid refreshing results while transitioning
         if (this.resultsRefreshInterval) {
             clearInterval(this.resultsRefreshInterval);
             delete this.resultsRefreshInterval;
         }
 
-        Promise.all([fadeOutPromise, nextQuestionPromise]).then(async function (results) {
```

Comment:
On Discord you asked me if we had something to prevent "multi-next" on the session.
I think the answer was "no" because we faded out immediately, so the user did not have the possibility to click multiple times.

Now that you're making that synchronous (we get the next question information and THEN start fading out), maybe it would be nice to prevent the multi-next?
I'm afraid server lags will make the button feel like it's not working during a big session.
At least previously it was fading out directly so you had that feeling of "oh ok it's doing something".

Disabling the button and adding a small spinner somewhere would be nice, I let you see if/how it's doable.

Reply:
Well the issue remains if you use navigation keys. Even if the button is not displayed anymore, you still can "press" go next using arrows. In survey form, I could not break the background. But here I could. Indeed we should probably avoid multi-next, but not only on the button itself. I thought about a debounce on "next question" method.

Reply:
We could force the display of the little arrow that turns into a spinning loading thing when you press the navigation key as well?

The idea would be to have a "don't worry, it's loading" feeling, and to avoid technical issues of course.